From 1cc2eb7375fa71359afd5c28a90951e2030930a9 Mon Sep 17 00:00:00 2001
From: Jin Yao <yao.jin@linux.intel.com>
Date: Mon, 12 May 2014 20:02:58 +0800
Subject: [PATCH] Change DW DMA suspend_noirq and resume_noirq callbacks to
 suspend_late and resume_early

we saw suspend process hang at here and never resume.

[   60.595667] PM: device INTL9C60:01 driver [dw_dmac] suspend noirq enter

Since DMA services for other components, we need to ensure DMA suspends
after the components and resume before the components are active.

Change-Id: Ifb8e64b6401a33ff30ec28a160961e88cb47d916
Signed-off-by: Jin Yao <yao.jin@linux.intel.com>
(cherry picked from commit 8b33146b6a20769ee029d89d6e8eeaeec33b1928)

Signed-off-by: Brett T. Warden <brett.t.warden@intel.com>
---
 drivers/dma/dw/platform.c |    8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/dma/dw/platform.c b/drivers/dma/dw/platform.c
index 9f61821..637d7ac 100644
--- a/drivers/dma/dw/platform.c
+++ b/drivers/dma/dw/platform.c
@@ -285,12 +285,8 @@ static int dw_resume_noirq(struct device *dev)
 #endif /* !CONFIG_PM_SLEEP */
 
 static const struct dev_pm_ops dw_dev_pm_ops = {
-	.suspend_noirq = dw_suspend_noirq,
-	.resume_noirq = dw_resume_noirq,
-	.freeze_noirq = dw_suspend_noirq,
-	.thaw_noirq = dw_resume_noirq,
-	.restore_noirq = dw_resume_noirq,
-	.poweroff_noirq = dw_suspend_noirq,
+	.suspend_late = dw_suspend_noirq,
+	.resume_early = dw_resume_noirq,
 };
 
 static struct platform_driver dw_driver = {
-- 
1.7.9.5

